### PR TITLE
support offline CPU in Linux

### PR DESCRIPTION
**Pull request by wangleis**
### Details:
 - *support offline CPU in Linux*
 - *Ignore SOC Ecore of MTL*
 - *enable Ecore of LNL*

### Tickets:
 - *CVS-154222*
 - *[issues-26889](https://github.com/openvinotoolkit/openvino/issues/26889)*

### CODE REVIEW

File: src/inference/src/os/lin/lin_system_conf.cpp [hunk 1]
```diff
@@ -201,7 +230,7 @@ CPU::CPU() {
 
     get_node_info_linux();
 
-    if (!get_cache_info_linux()) {
+    if (!get_info_linux(1)) {
```
Comment (sunxiaoxia2022) [inline src/inference/src/os/lin/lin_system_conf.cpp:233]: I think if we can change `int` value to `enum` which the meaning is more clearer.

Comment (wangleis) [reply src/inference/src/os/lin/lin_system_conf.cpp:233]: updated

**Comment by dmitry-gorokhov** [comment]
Please revert changes in oneDNN GPU submodule

**Comment by dmitry-gorokhov** [comment]
I am not very familiar with lin_system_conf.cpp impl. so would ask @sunxiaoxia2022 to review in details.

**Comment by wangleis** [comment]
> Please revert changes in oneDNN GPU submodule

@dmitry-gorokhov updated.

File: src/inference/src/os/lin/lin_system_conf.cpp [hunk 2]
```diff
@@ -23,76 +23,108 @@ CPU::CPU() {
...
+                                                     file_name[n]);
+                            if (cache_file.is_open()) {
+                                std::getline(cache_file, one_info);
+                            } else {
+                                if ((cpu_index == core_1) && (n == 2)) {
+                                    system_info_table.clear();
+                                    return -1;
+                                }
+                            }
+                            system_info_table[cpu_index][n] = one_info;
```
Comment (sunxiaoxia2022) [inline src/inference/src/os/lin/lin_system_conf.cpp:118]: Why not add std::move() like line 102?

Comment (wangleis) [reply src/inference/src/os/lin/lin_system_conf.cpp]: updated

**Review by sunxiaoxia2022 — APPROVED** [review]

**Comment by sunxiaoxia2022** [comment]
@dmitry-gorokhov It is ok now from my side.

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**